core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/internal/CarbonUIServiceComponent.java

```diff
@@ -202,6 +178,32 @@ protected void activate(ComponentContext ctxt) {
         }
     }
 
+    private void printAdditionalContext(Context additionalContext, String adminConsoleURL, String webContextRoot) {
+        if ("/".equals(webContextRoot)) {
+            webContextRoot = "";
+        }
+        if (additionalContext != null && !"".equals(additionalContext.getContextName()) &&
+            !"null".equals(additionalContext.getContextName())) {
+            // Adding the other context url
+            int index = adminConsoleURL.lastIndexOf("carbon");
+            String defContextUrl = adminConsoleURL.substring(0, index);
+            if (!"".equals(webContextRoot)) {
```

Comment:
Shall we change this to !webContextRoot.isEmpty() ?

Reply:
That will require an additional null check. Since this is in startup process IMHO we can skip that bull check with this approach. WDYT?

Reply:
Here, what will happen if webContextRoot is null ? wouldn't it be good to check for null as well ?

Reply:
AFAIK if WebContextRoot(in carbon,xml) is null then we assume that as / in other places.

Reply:
Input validation is a must. If NULL is not accepted, better checking for it.